scripts/CEE/hive-data-collection/script.sh

```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+#
+# Collect data from Hive clusters in order to troubleshoot/debug cluster provisioning delays.
+# Input:
+#  (Env) CLUSTERID
+#        Description: The Internal Cluster ID for which data is being collected
+
+set -e
+set -o nounset
+set -o pipefail
+
+start_job(){
+    echo "Job started at $(date +"%Y-%m-%d %T")"
+    echo ".................................."
+    echo
+}
+
+finish_job(){
+    echo
+    echo ".................................."
+    echo "Job finished at $(date +"%Y-%m-%d %T")"
+}
+
+HIVE_NAMESPACE="uhc-production-${CLUSTER_ID}"
+
+if ! oc get namespace "${HIVE_NAMESPACE}" &>/dev/null; then
+    echo "Namespace ${HIVE_NAMESPACE} does not exist. Please check the CLUSTER_ID."
+    exit 1
+fi
+
+## Function which checks the clusterdeployment status
+## and print the details of each clusterdeployment
+clusterdeployment(){
+       
+    echo "### Checking clusterdeployment status..."
+    echo
+    oc get clusterdeployment -n "${HIVE_NAMESPACE}"
+    echo
+    oc get clusterdeployment -n "${HIVE_NAMESPACE}" -o jsonpath='{.items[*].metadata.name}' | tr ' ' '\n' | sort -u | while read -r cluster; do
+        echo "### ClusterDeployment: ${cluster}"
+        echo "------------------------------------------------------------------"
+        echo
+        oc get clusterdeployment "${cluster}" -n "${HIVE_NAMESPACE}" -o yaml
+        echo
+    done 
+}
+
+## Function which checks pods status
+## and print the details of each pod
+pods(){
+
+    echo "### Checking pods status..."
+    echo
+    oc get pods -n "${HIVE_NAMESPACE}" -o wide
+    echo
+    oc get pods -n "${HIVE_NAMESPACE}" -o jsonpath='{.items[*].metadata.name}' | tr ' ' '\n' | sort -u | while read -r pod; do
```

Comment:
Can we just execute `oc describe po -n $NAMESPACE` here if we would like to describe all the pods?